core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -437,6 +505,68 @@ public PrivateKey getDefaultPrivateKey() throws Exception {
                 "available only for the super tenant.");
     }
 
+    /**
+     * This method loads the private key of a given tenant key store.
+     *
+     * @param keyStoreName  Name of the tenant key store.
+     * @param alias         Alias of the private key.
+     * @return Private key corresponding to the alias.
+     */
+    private PrivateKey getTenantPrivateKey(String keyStoreName, String alias) throws Exception {
+
+        String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName;
+        org.wso2.carbon.registry.api.Resource resource;
+        KeyStore keyStore;
+
+        if (registry.resourceExists(path)) {
+            resource = registry.get(path);
+        } else {
+            throw new SecurityException("Given Key store is not available in registry : " + keyStoreName);
+        }
+
+        CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
+        String encryptedPassword = resource
+                .getProperty(RegistryResources.SecurityManagement.PROP_PRIVATE_KEY_PASS);
+        String privateKeyPasswd = new String(cryptoUtil.base64DecodeAndDecrypt(encryptedPassword));
+
+        if (isCachedKeyStoreValid(keyStoreName)) {
+            keyStore = tenantKeyStores.get(keyStoreName).getKeyStore();
+        } else {
+            byte[] bytes = (byte[]) resource.getContent();
+            String keyStorePassword = new String(cryptoUtil.base64DecodeAndDecrypt(resource.getProperty(
+                    RegistryResources.SecurityManagement.PROP_PASSWORD)));
+            keyStore = KeyStore.getInstance(resource
+                    .getProperty(RegistryResources.SecurityManagement.PROP_TYPE));
+            ByteArrayInputStream stream = new ByteArrayInputStream(bytes);
+            keyStore.load(stream, keyStorePassword.toCharArray());
+
+            KeyStoreBean keyStoreBean = new KeyStoreBean(keyStore, resource.getLastModified());
+            updateKeyStoreCache(keyStoreName, keyStoreBean);
+
```

Comment:
Remove additional line

Reply:
Fixed in https://github.com/wso2/carbon-kernel/pull/4025/commits/6ea03c474d19b5dd7c6138a60ffb49031621d383